src/cases/containers/query-management-container/query-management-container.component.ts

```diff
@@ -341,23 +341,20 @@ export class QueryManagementContainerComponent implements OnInit, OnDestroy {
       this.caseNotifier.caseView,
       this.featureToggleService.getValue(this.LD_QUALIFYING_QUESTIONS, [])
     ]).pipe(
-      map(([caseView, caseTypeQualifyingQuestions]: [CaseView, CaseTypeQualifyingQuestions[]]) => {
-        this.caseId = caseView.case_id;
+      map(([caseView, caseTypeQualifyingQuestions]) => {
+        const caseId = caseView.case_id;
 
         // Safely access the qualifying questions for the current case type
-        const qualifyingQuestions: QualifyingQuestion[] = caseTypeQualifyingQuestions?.[caseView.case_type.id] || [];
+        const qualifyingQuestions = (caseTypeQualifyingQuestions?.[caseView.case_type.id] || []).map((question) => {
+          // Interpolate ${[CASE_REFERENCE]} in all qualifying questions
+          const url = question.url?.replace('${[CASE_REFERENCE]}', caseId);
+          const markdown = question.markdown?.replace('${[CASE_REFERENCE]}', caseId);
```

Comment:
You almost certainly need to replace all, probably in both scenarios, To do that you'll have to convert the placeholder to a regex with the g modifier (global), something like
const placeHolder = /\$\[\{CASE_REFERENCE\]\}/g
const markdown = question.markdown?.replace(placeHolder, caseId);